src/opera_disp_tms/generate_sw_disp_tile.py

```diff
@@ -0,0 +1,252 @@
+import argparse
+import warnings
+from dataclasses import dataclass
+from datetime import datetime
+from pathlib import Path
+from typing import Iterable, Tuple
+
+import numpy as np
+import xarray as xr
+from osgeo import gdal
+from rasterio.transform import Affine
+
+from opera_disp_tms import utils
+from opera_disp_tms.find_california_dataset import Granule, find_california_dataset
+from opera_disp_tms.s3_xarray import open_opera_disp_granule
+from opera_disp_tms.utils import DATE_FORMAT
+
+
+gdal.UseExceptions()
+
+
+@dataclass
+class FrameMeta:
+    """Dataclass for frame metadata"""
+
+    frame_id: int
+    reference_date: datetime
+    reference_point_array: tuple  # column, row
+    reference_point_geo: tuple  # lon, lat
+
+
+def extract_frame_metadata(frame_metadata: dict[str, str], frame_id: int) -> FrameMeta:
+    """Extract frame metadata from a frame metadata dictionary
+
+    Args:
+        frame_metadata: A dictionary of attributes for multiple frames
+        frame_id: the ID of the desired frame
+
+    Returns:
+        A FrameMeta object for the desired frame
+    """
+    ref_date = datetime.strptime(frame_metadata[f'FRAME_{frame_id}_REF_TIME'], DATE_FORMAT)
+
+    ref_point_array = frame_metadata[f'FRAME_{frame_id}_REF_POINT_ARRAY'].split(', ')
+    ref_point_array = tuple(int(x) for x in ref_point_array)
+
+    ref_point_geo = frame_metadata[f'FRAME_{frame_id}_REF_POINT_GEO'].split(', ')
+    ref_point_geo = tuple(float(x) for x in ref_point_geo)
+
+    return FrameMeta(frame_id, ref_date, ref_point_array, ref_point_geo)
+
+
+def frames_from_metadata(metadata_path: Path) -> dict[int, FrameMeta]:
+    """Extract frame metadata from a metadata GeoTiff file
+
+    Args:
+        metadata_path (Path): Path to the metadata GeoTiff file
+
+    Returns:
+        Dictionary of frame metadata indexed by frame id
+    """
+    info_dict = gdal.Info(str(metadata_path), format='json')
+    frame_metadata = info_dict['metadata']['']
+    frame_ids = [int(x) for x in frame_metadata['OPERA_FRAMES'].split(', ')]
```

Comment:
Do we want to (now or later) handle items with no `frame_metadata['OPERA_FRAMES']`? Even just outputting a list of which frames/tiles aren't generated could be helpful!

Reply:
I think we should not create the metadata tile in this case, so generate_sw_disp_tile.py shouldn't have to handle it.